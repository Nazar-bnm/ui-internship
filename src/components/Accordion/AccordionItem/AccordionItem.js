import React from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';
import './AccordionItem.scss';

const CN = 'accordion-item';

const AccordionItem = ({
  hide,
  onClick,
  title,
  description,
  height,
  scroll,
}) => {
  const heightStyle = {
    height: height,
  };

  return (
    <div className={CN}>
      <div
        className={cx(`${CN}__header`, { [`${CN}__header--active`]: !hide })}
        onClick={onClick}
      >
        <h4>{title}</h4>
        <i className={cx('caret', 'right', 'icon', { 'icon--down': !hide })} />
      </div>

      <div
        className={cx([`${CN}__description`], {
          [`${CN}__description--hide`]: hide,
          [`${CN}__description--show`]: !hide,
          scroll: scroll,
        })}
        style={!hide ? heightStyle : {}}
      >
        {description}
      </div>
    </div>
  );
};

AccordionItem.propTypes = {
  hide: PropTypes.bool,
  onClick: PropTypes.func,
  title: PropTypes.string,
  description: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  height: PropTypes.string,
  scroll: PropTypes.bool,
};

AccordionItem.defaultProps = {
  hide: true,
  onClick: () => {},
  title: '',
  description: '',
  height: '',
  scroll: false,
};

export default AccordionItem;
