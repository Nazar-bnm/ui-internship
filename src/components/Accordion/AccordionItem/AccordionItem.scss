@import '../../../scss/global_styles';

.accordion-item {
  font-family: font(secondary);

  &__header {
    transition: color linear 0.5s;
    display: flex;
    justify-content: space-between;
    align-items: center;
    border-top: 1px solid $silver-chalice;
    padding: 1rem 0;
    cursor: pointer;
    font: 0.9rem font(secondary);
    text-transform: uppercase;

    &:hover,
    &:active,
    &--active {
      color: $silver-chalice;
    }

    i.icon {
      transition: transform linear 0.5s;
      margin: 0;

      &--down {
        transform: rotate(90deg);
      }
    }
  }

  &__description {
    &--hide,
    &--show {
      overflow-y: hidden;
      transition: height linear 0.5s;
      height: 0;
    }

    &--show {
      height: 100px;
    }
  }

  .scroll {
    overflow-y: scroll;

    &::-webkit-scrollbar {
      width: 5px;
    }

    &::-webkit-scrollbar-track,
    &::-webkit-scrollbar-thumb {
      border-radius: 10px;
      border: 1px solid $silver-chalice;
      background-color: $silver-chalice;
    }

    &::-webkit-scrollbar-thumb {
      background: $white;
    }
  }
}
