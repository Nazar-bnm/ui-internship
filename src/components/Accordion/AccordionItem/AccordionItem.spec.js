import React from 'react';
import { shallow } from 'enzyme';
import AccordionItem from './AccordionItem';

describe('<AccordionItem />', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      title: 'hello',
      description: 'Architecto, earum temporibus quidem ex eaque unde harum sit, deleniti tempore',
      hide: true,
    };

    wrapper = shallow(<AccordionItem {...props} />);
  });

  it('should match snapshot <AccordionItem />', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke function mockClickHandler when event on click occurs', () => {
    const mockClickHandler = jest.fn();

    wrapper = shallow(<AccordionItem {...props} onClick={mockClickHandler}/>);
    wrapper.find('.accordion-item-header').first().simulate('click');

    expect(mockClickHandler.mock.calls.length).toBe(1);
  });
});
