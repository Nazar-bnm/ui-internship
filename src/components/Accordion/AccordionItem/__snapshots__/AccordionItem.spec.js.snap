// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionItem /> should match snapshot <AccordionItem /> 1`] = `
<div
  className="accordion-item"
>
  <div
    className="accordion-item__header"
    onClick={[Function]}
  >
    <h4>
      hello
    </h4>
    <i
      className="caret right icon"
    />
  </div>
  <div
    className="accordion-item__description accordion-item__description--hide"
    style={Object {}}
  >
    Architecto, earum temporibus quidem ex eaque unde harum sit, deleniti tempore
  </div>
</div>
`;
